core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -506,6 +514,44 @@ private boolean isThisGraalVMRCObsolete() {
         return false;
     }
 
+    private static String getLinuxID(String option) {
+        Process process = null;
+        BufferedReader reader = null;
+
+        try {
+            StringBuilder responseBuilder = new StringBuilder();
+            String line;
+
+            ProcessBuilder idPB = new ProcessBuilder().command("id", option);
+            idPB.redirectError(new File("/dev/null"));
+
+            try {
+                process = idPB.start();
+                process.getOutputStream().close();
+                reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+                while ((line = reader.readLine()) != null) {
+                    responseBuilder.append(line);
+                }
+
+                return responseBuilder.toString();
+
+            } finally {
+                if (process != null) {
+                    process.getInputStream().close();
```

Comment:
This close could throw an exception, leaving the error stream and reader open.  Also, the fact that each close operation could thrown an exception violates the principle of "never throw from `finally`".

Reply:
If try-with-resources is used and `/dev/null` feeds the input as well as output, then the `finally` block is not needed anymore.